Suite2p/suite2p_main.py

```diff
@@ -0,0 +1,205 @@
+    f_cells = f_cells[iscell, :]
+    spks    = spks[iscell, :]
+     
+    n_cells = len(stats)
+    Ly = output_ops["Ly"]
+    Lx = output_ops["Lx"]
+
+    #FootPrint to use to save the ROIs contour in doric
+    footPrint = np.zeros((n_cells, Ly, Lx))
+    for i, stat in enumerate(stats):
+        footPrint[i, stat['ypix'] - int(stat['iplane']/2) * Ly, stat['xpix'] - (stat['iplane'] % 2) * Lx] = 1
+
+    print(s2p_defs.Messages.ROI_NAMES, flush = True)
+    ids           = [i + 1 for i in range(n_cells)]
+    dataset_names = [defs.DoricFile.Dataset.ROI.format(str(id_).zfill(4)) for id_ in ids]
+    usernames     = [defs.DoricFile.Dataset.ROI.format(id_) for id_ in ids]
+
+    with h5py.File(doricFileName, 'a') as f:
+        # Check if Suite2p results already exist
+        operationCount = utils.operation_count(vpath, f, s2p_defs.DoricFile.Group.ROISIGNALS, params_doric, params_source)
+
+        params_doric[defs.DoricFile.Attribute.Group.OPERATIONS] += operationCount
+
+        print(s2p_defs.Messages.SAVING_ROIS, flush=True)
+        rois_grouppath   = f"{vpath}/{s2p_defs.DoricFile.Group.ROISIGNALS+operationCount}"
+        rois_seriespath  = f"{rois_grouppath}/{series}"
+        attrs = {"RangeMin": 0, "RangeMax": 0, "Unit": "AU"}
+        save_roi_signals(f_cells, footPrint, time_, f, rois_seriespath, sensor,
+                         ids            = ids,
+                         dataset_names  = dataset_names,
+                         usernames      = usernames,
+                         attrs          = attrs,
+                         planeID        = [stat['iplane'] + 1 for stat in stats])
+        utils.save_attributes(utils.merge_params(params_doric, params_source), f, rois_grouppath)
+        for planeSensor in f[rois_seriespath].keys():
+            utils.print_group_path_for_DANSE(f"{rois_seriespath}/{planeSensor}")
+
+        print(s2p_defs.Messages.SAVING_SPIKES, flush=True)
+        spikes_grouppath   = f"{vpath}/{s2p_defs.DoricFile.Group.SPIKES+operationCount}"
+        spikes_seriespath  = f"{spikes_grouppath}/{series}"
+        attrs = {"RangeMin": 0, "RangeMax": 0, "Unit": "AU"}
+        spikes = correctSpikesValues(spks, f_cells)
+        save_spikes(spikes, time_, f, spikes_seriespath , sensor,
+                            dataset_names  = dataset_names,
+                            usernames      = usernames,
+                            attrs          = attrs,
+                            planeID        = [stat['iplane'] + 1 for stat in stats])
+        utils.save_attributes(utils.merge_params(params_doric, params_source), f, spikes_grouppath)
+        for planeSensor in f[spikes_seriespath].keys():
+            utils.print_group_path_for_DANSE(f"{spikes_seriespath}/{planeSensor}")
+
+
+def correctSpikesValues(spks:np.ndarray, f_cells:np.ndarray)->np.ndarray:
+        spikes: np.ndarray = (spks/spks) * f_cells
+        spikes[np.isnan(spikes)] = 0
+
+        return spikes
+
+
+def save_roi_signals(
```

Comment:
Can't we use the utilities `save_roi_signals` and `save_spikes`?

Reply:
We need to have the Plane ID in the groups where the signal are save so it's really different than the one in utilities that is why we copy them there and edit them too (We try to use the one in utilities first)

Reply:
Could you make some changes above, so that it uses the functions from utils if number of planes is 1?
What other changes do we have to make if we have only 1 plane?

Reply:
oki

Reply:
I think I can just remove the plane ID in the function and just loop on each plane and call the `save_roi_signals` from util like that I think I don't have to edit the save_roi_signals